src/main/java/Arona/Parser.java

```diff
@@ -0,0 +1,162 @@
+package Arona;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class Parser {
+    public Parser() {}
+
+    /**
+     * Handles processing of input by calling the necessary classes
+     * @param  storage  should only be initiated once at the start of the programme
+     * @param  taskList  should only be initiated once at the start of the programme
+     * @param  ui  should only be initiated once at the start of the programme
+     */
+    public static void parse(String input, Storage storage, TaskList taskList, Ui ui) throws Exception {
+
+        // Bye command
+        if (input.equalsIgnoreCase("bye")) {
+            // Save task list to data.txt
+            storage.save(taskList);
+
+            // Reply
+            ui.showFarewell();
+        }
+
+        // Debug: print data location
+        else if (input.equalsIgnoreCase("storage")) {
+            ui.showStorage(storage);
+        }
+
+        // List command
+        else if (input.equalsIgnoreCase("list")) {
+            ui.showList(taskList);
+        }
+
+        // Delete command
+        else if (input.toLowerCase().startsWith("delete ")) {
+
+            // Extract and save task number
+            String[] data = input.split("delete", 2);
+
+            // Check if number
+            if (!data[1].stripLeading().matches("^[1-9]\\d*$")) {
+                throw new AronaException("Error! Please input a positive number.");
+            }
+
+            // Save index as integer
+            int index = Integer.parseInt(data[1].stripLeading());
+
+            // Process
+            Task task = taskList.remove(index - 1);
+
+            // Reply
+            ui.showDelete(taskList.size(), task);
+        }
+
+        // Mark and unmark command
+        else if (input.toLowerCase().startsWith("mark ") || input.toLowerCase().startsWith("unmark ")) {
+
+            // Extract and save task number
+            String[] data = input.split("mark", 2);
+
+            // Extract and save action
+            boolean action = input.toLowerCase().startsWith("mark");
+
+            // Check if number
+            if (!data[1].stripLeading().matches("^[1-9]\\d*$")) {
+                throw new AronaException("Error! Please input a positive number.");
+            }
+
+            // Save index as integer
+            int index = Integer.parseInt(data[1].stripLeading());
+
+            // Process
+            Task task = taskList.setStatus(index - 1, action);
+
+            // Reply
+            ui.showMark(task, action);
+        }
+
+        // Todos or events or deadline command
+        else if (input.toLowerCase().startsWith("todo ") || input.toLowerCase().startsWith("event ") || input.toLowerCase().startsWith("deadline ")) {
```

Comment:
This line looks to be over 120 characters for the coding standard limit, would recommend breaking this into a new line